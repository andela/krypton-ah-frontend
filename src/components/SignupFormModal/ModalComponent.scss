@import "../colors";
@import "../fonts";

body {
  font: {
    family: "roboto";
    weight: 100;
  }
}

.ui {

	&.tiny {
	  &.modal {
			border: 1px solid $primarycolor;
			border-radius: 0.938rem;
			margin: 2rem;
			padding: 1.25rem;
      width: 28rem;
    }
  }
}

.create-account span {
  text-decoration: underline;
}

#signup-button {
  border-radius: 0;
  color: $black !important;
  font-family: roboto, arial, 'sans-serif';
  font-weight: 700;
  margin: 3.125rem 0;

  &:hover {
    background: $primarycolor !important;
    color: $white !important;
    transition: 0.5s ease-in-out;
  }
}
