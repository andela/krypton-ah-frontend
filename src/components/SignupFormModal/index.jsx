import React from 'react';
import {
  Button,
  Modal,
  Header,
  Form,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './ModalComponent.scss';
import ModalHeader from '../ModalHeader';
import SocialIcons from '../SocialComponent';
import ModalDivider from '../ModalDivider';
import ModalFooter from '../ModalFooter';

function SignUpModal({ onChange, onClick }) {
  return (
    <div>
      <Modal
        size="tiny"
        trigger={(
          <Button className="ui button black">
          Write An Article
          </Button>
        )}>
        <Modal.Content>
          <ModalHeader title="sign up" />
          <Modal.Description>
            <Header />
            <Form id="signUp">
              <Form.Group widths="equal">
                <Form.Input fluid placeholder="Firstname" onChange={onChange} required />
                <Form.Input fluid placeholder="Lastname" onChange={onChange} required />
              </Form.Group>
              <Form.Group widths="equal">
                <Form.Input fluid placeholder="Email Address" onChange={onChange} required />
              </Form.Group>
              <Form.Group widths="equal">
                <Form.Input fluid placeholder="Password" onChange={onChange} required />
              </Form.Group>
              <Button onClick={onClick} type="submit" basic fluid huge="true" id="signup-button">Sign Up</Button>
            </Form>
            <ModalDivider title="Or sign up with" />
            <SocialIcons />
            <ModalFooter title="Already have an account?" link="Sign In" />
          </Modal.Description>
        </Modal.Content>
      </Modal>
    </div>
  );
}

SignUpModal.propTypes = {
  onChange: PropTypes.func,
  onClick: PropTypes.func
};
SignUpModal.defaultProps = {
  onChange: PropTypes.func,
  onClick: PropTypes.func
};

export default SignUpModal;
