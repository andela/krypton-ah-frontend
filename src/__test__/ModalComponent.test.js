import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import SignupFormModal from '../components/SignupFormModal';

function checkModalComponent() {
  test('it should matches the snapshot', () => {
    const component = create(<SignupFormModal />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('should renders 1 <Button /> component', () => {
    const wrapper = shallow(<SignupFormModal />);
    expect(wrapper.find('Button')).toHaveLength(1);
  });
  test('should render without crashing', () => {
    const wrapper = shallow(<SignupFormModal />);
    expect(wrapper).toBeTruthy();
  });
}

describe('SignupFormModal Component', () => {
  checkModalComponent();
});
