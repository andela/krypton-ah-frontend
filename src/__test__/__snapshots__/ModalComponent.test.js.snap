// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormModal Component it should matches the snapshot 1`] = `
<div>
  <button
    className="ui button ui button black"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Write An Article
  </button>
</div>
`;
